sentry_youtrack/forms.py

```diff
@@ -168,43 +169,59 @@ class YouTrackConfigurationForm(forms.Form):
-
         self.client_errors = {}
-        client = None
-        initial = kwargs.get("initial")
 
+        initial = kwargs.get("initial")
         if initial:
             client = self.get_youtrack_client(initial)
-            if not client and not args[0]:
+            if client:
+                if initial.get('project'):
+                    choices = self.get_ignore_field_choices(
+                        client, initial.get('project'))
+                    self.fields['ignore_fields'].choices = choices
+
+                choices = self.get_project_field_choices(
+                    client, initial.get('project'))
+                self.fields["project"].choices = choices
+
+                if not any(args) and not initial.get('project'):
+                    self.second_step_msg = _(
+                        "Your credentials are valid but plugin is NOT active "
+                        "yet. Please fill in remaining required fields.")
+
+            if self.client_errors and not args[0]:
                 self.full_clean()
                 for field, error in self.client_errors.items():
                     self._errors[field] = [error]
-
-        default_fields = ['url', 'username', 'password']
-        if initial and client:
-            default_fields.append('project')
-
-        if initial and client:
-            if initial.get('project'):
-                fields = client.get_project_fields_list(initial.get('project'))
-                names = [field['name'] for field in fields]
-                self.fields['ignore_fields'].choices = zip(names, names)
-
-            projects = [(' ', u"- Choose project -")]
-            for project in client.get_projects():
-                display = "%s (%s)" % (project['name'], project['id'])
-                projects.append((project['id'], display))
-            self.fields["project"].choices = projects
-
-            if not any(args) and not initial.get('project'):
-                self.second_step_msg = _("Your credentials are valid but "
-                                         "plugin is NOT active yet. Please "
-                                         "fill in remaining required fields.")
         else:
             del self.fields["project"]
             del self.fields["default_tags"]
             del self.fields["ignore_fields"]
 
+    def get_ignore_field_choices(self, client, project):
+        try:
+            fields = list(client.get_project_fields_list(project))
+        except HTTPError:
+            self.client_errors['project'] = self.error_message[
+                'invalid_project'] % (project,)
+        else:
+            names = [field['name'] for field in fields]
+            return zip(names, names)
+        return []
+
+    def get_project_field_choices(self, client, project):
+        choices = [(' ', u"- Choose project -")]
+        try:
+            projects = list(client.get_projects())
+        except HTTPError:
```

Comment:
We want to handle only the case when the current project is no longer available. `TypeError` indicates problems with API.